bin/test-without-stubs

```diff
@@ -20,6 +22,11 @@ for dir in exercises/*; do
 done
 
 go test -cpu 2 $RACE ./...
+if [ $? -ne 0 ]; then
+  # The test has failed, but don't immediately exit yet!
+  # We still need to run the cleanup code below!
```

Comment:
since we're in the go repo, this really makes you think "Wow, if only we had `defer` here! Then we could have just `defer`red the cleanup function at the beginning, exited immediately here, and the cleanup function would automatically run.

Turns out we can in fact do such a thing in a shell script with traps... so I may try that.